tests/unit/customizations/s3/test_s3handler.py

```diff
@@ -903,190 +904,124 @@ def test_remove_bucket(self):
-        mock_task_class = mock.Mock()
-        s3handler._enqueue_upload_single_part_task(
-            part_number=1, chunk_size=2, upload_context=None,
-            filename=None, task_class=mock_task_class,
-            payload=b'This is a test'
-        )
-        args, kwargs = mock_task_class.call_args
-        self.assertIn('payload', kwargs.keys())
-        self.assertEqual(kwargs['payload'], b'This is a test')
-
-    def test_enqueue_multipart_download_stream(self):
-        """
-        This test ensures the right calls are made in ``_enqueue_tasks()``
-        if the file should be a multipart download.
-        """
-        s3handler = S3StreamHandler(
-            self.session, self.params,
-            runtime_config=runtime_config(multipart_threshold=5))
-        s3handler.executor = mock.Mock()
-        fileinfo = FileInfo('filename', operation_name='download',
-                            is_stream=True)
-        with mock.patch('awscli.customizations.s3.s3handler'
-                        '.S3StreamHandler._enqueue_range_download_tasks') as \
-                mock_enqueue_range_tasks:
-            with mock.patch('awscli.customizations.s3.fileinfo.FileInfo'
-                            '.set_size_from_s3') as mock_set_size_from_s3:
-                # Set the file size to something larger than the multipart
-                # threshold.
-                fileinfo.size = 100
-                # Run the main enqueue function.
-                s3handler._enqueue_tasks([fileinfo])
-                # Assert that the size of the ``FileInfo`` object was set
-                # if we are downloading a stream.
-                self.assertTrue(mock_set_size_from_s3.called)
-                # Ensure that this download would have been a multipart
-                # download.
-                self.assertTrue(mock_enqueue_range_tasks.called)
-
-    def test_enqueue_range_download_tasks_stream(self):
-        s3handler = S3StreamHandler(
-            self.session, self.params,
-            runtime_config=runtime_config(multipart_chunksize=100))
-        s3handler.executor = mock.Mock()
-        fileinfo = FileInfo('filename', operation_name='download',
-                            is_stream=True, size=100)
-        s3handler._enqueue_range_download_tasks(fileinfo)
-        # Ensure that no request was sent to make a file locally.
-        submitted_tasks = s3handler.executor.submit.call_args_list
-        self.assertNotEqual(type(submitted_tasks[0][0][0]),
-                            CreateLocalFileTask)
+        expected_size = 6
+        self.params['expected_size'] = expected_size
+        handler = S3TransferStreamHandler(self.session, self.params)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        self.assertEqual(self.transfer_future.meta.size, expected_size)
```

Comment:
In S3Transfer you mean? I don't think it makes sense to test functionality of a separate library.